psycopg_c/psycopg_c/_psycopg/transform.pyx

```diff
@@ -440,50 +445,56 @@ cdef class Transformer:
         cdef PGresAttValue *attval
         cdef object record  # not 'tuple' as it would check on assignment
 
-        cdef object records = PyList_New(row1 - row0)
-        for row in range(row0, row1):
-            record = PyTuple_New(self._nfields)
-            Py_INCREF(record)
-            PyList_SET_ITEM(records, row - row0, record)
+        cdef int rowcount = row1 - row0
+        cdef object records = PyList_New(rowcount)
 
         cdef PyObject *loader  # borrowed RowLoader
         cdef PyObject *brecord  # borrowed
+        cdef int advance
         row_loaders = self._row_loaders  # avoid an incref/decref per item
 
-        for col in range(self._nfields):
-            loader = PyList_GET_ITEM(row_loaders, col)
-            if (<RowLoader>loader).cloader is not None:
-                for row in range(row0, row1):
-                    brecord = PyList_GET_ITEM(records, row - row0)
-                    attval = &(ires.tuples[row][col])
-                    if attval.len == -1:  # NULL_LEN
-                        pyval = None
-                    else:
-                        pyval = (<RowLoader>loader).cloader.cload(
-                            attval.value, attval.len)
-
-                    Py_INCREF(pyval)
-                    PyTuple_SET_ITEM(<object>brecord, col, pyval)
-
-            else:
-                for row in range(row0, row1):
-                    brecord = PyList_GET_ITEM(records, row - row0)
-                    attval = &(ires.tuples[row][col])
-                    if attval.len == -1:  # NULL_LEN
-                        pyval = None
-                    else:
-                        b = PyMemoryView_FromObject(
-                            ViewBuffer._from_buffer(
-                                self._pgresult,
-                                <unsigned char *>attval.value, attval.len))
-                        pyval = PyObject_CallFunctionObjArgs(
-                            (<RowLoader>loader).loadfunc, <PyObject *>b, NULL)
+        while row0 < row1:
+            # TODO: make advance configurable
+            advance = 100 if row1 - row0 > 100 else row1 - row0
+            for row in range(advance):
+                record = PyTuple_New(self._nfields)
+                Py_INCREF(record)
+                PyList_SET_ITEM(records, row, record)
```

Comment:
`row` always go from 0 to 99 here, it is expected to go from 0 to rowcount - 1.

How about introducing 2 vars: pr0 and pr1 as page row 0 and 1, and so iterate first on the page, then on the page record?

```
for pr0 in range(row0, row1, 100):
    pr1 = min(pr0 + 100, row1)
    for row in range(pr0, pr1):
        ...
```

Reply:
Woops, yeah thats a mistake :sweat_smile: